Add Concurrency Kit checks and define setting to configure.ac

ArcEye opened the pull request:
Part of the multicore work which is missing for some reason.

Fixes #221 

Signed-off-by: Mick <arceye@mgware.co.uk>

ArcEye committed:
Add Concurrency Kit checks and define setting to configure.ac

Part of the multicore work which is missing for some reason.

Signed-off-by: Mick <arceye@mgware.co.uk>

ArcEye committed:
Change configure.ac to fail in absence of concurrencykit lib

RIP builds can function with or without it, but packages
will have problems once dockers contain libtk0 and build
against it.

Signed-off-by: Mick <arceye@mgware.co.uk>

ArcEye committed:
Add dependencies for concurrencykit.

libck-dev dep for builds, libck0 dep for system install.

Signed-off-by: Mick <arceye@mgware.co.uk>

zultron commented on debian/control.d/control.buster.in:
```diff
@@ -4,4 +4,4 @@ Depends: libavahi-client3 (>= 0.6.16), libavahi-common3 (>= 0.6.16), libc6 (>= 2
     libboost-thread1.62.0 | libboost-thread1.67.0,
     bc, cgroup-tools, libtk-img (>= 1.13), procps, psmisc, python-tk, python-numpy, python-xlib,
-    python-gtkglext1, python-configobj, python-zmq, python-protobuf (>= 2.4.1), python-avahi, python-simplejson, python-pyftpdlib, tclreadline, tcl8.6, tk8.6, uuid-runtime, yapps2
+    python-gtkglext1, python-configobj, python-zmq, python-protobuf (>= 2.4.1), python-avahi, python-simplejson, python-pyftpdlib, tclreadline, tcl8.6, tk8.6, uuid-runtime, yapps2, libck0
```
There's no `libck0` package in Buster.

ArcEye replied:
Don't know what is going on with Debian at present.

This situation is the same for quite a few packages at present, if it is in sid, why don't they just backport it, instead of leaving Buster with no package?

I am actually using sid presently so did not even think to check if Buster was covered.

Does your `/etc/apt/sources.list.d` in the dockers have a `machinekit.list` file?
I am very tempted to just build a `libck0 / libck-dev` packages for Buster and host it in the repo, rather than have the stupid situation where people running the most up to date distro, cannot use the ck features.

zultron replied:
Yes, this is dumb.  I'll ask around the Debian community at some point.

The mk-cross-builder images do configure the MK apt repository.  If you add that package, let me know and I'll update the images.

Did you see my comment in #221, `libck` is also not built for Jessie i386?

zultron commented on debian/control.d/control.in:
```diff
@@ -18,5 +18,5 @@ Build-Depends: debhelper (>= 6),
     python-simplejson, libboost-thread-dev, libcgroup-dev,
     yapps2-runtime | python-yapps, yapps2,
-    python-pyftpdlib, @BUILD_DEPS@
+    python-pyftpdlib, libck-dev, @BUILD_DEPS@
```
There's no `libck-dev` package in Buster, so this needs to be moved into `debian/configure` and added through the `BUILD_DEPS` variable.

zultron commented on debian/control.d/control.jessie.in:
```diff
@@ -3,4 +3,4 @@ Depends: libavahi-client3 (>= 0.6.16), libavahi-common3 (>= 0.6.16), libc6 (>= 2
     zlib1g (>= 1:1.1.4), python (<< 2.8), python (>= 2.7~), python:any (>= 2.7.5-5~), libboost-python1.55.0, libboost-thread1.55.0, libboost-system1.55.0,
     bc, cgroup-tools, libtk-img (>=1.13), procps, psmisc, python-tk, python-numpy, python-xlib, python-gtkglext1, python-configobj, python-zmq, python-protobuf (>= 2.4.1),
-    python-avahi, python-simplejson, python-pyftpdlib, tclreadline, tcl8.6, tk8.6, uuid-runtime, yapps2-runtime
+    python-avahi, python-simplejson, python-pyftpdlib, tclreadline, tcl8.6, tk8.6, uuid-runtime, yapps2-runtime, libck0
```
`dh_shlibdeps` should be able to pick this up automatically, and this can be dropped.  Same for the hunk in `control.stretch.in`.

zultron reviewed (changes requested):
Thanks for the PR.  Buster doesn't have a `libck-dev` package at all, AFAICT (but Sid does?!?), so that needs fixing.  The other comment about `libck0` being picked up by `dh_shlibdeps` is just a nit.

zultron commented on src/configure.ac:
```diff
@@ -976,10 +976,24 @@ AC_ARG_ENABLE(dev,
+	# needed for kbuild:
+        CK_INCLUDE=`pkg-config --cflags-only-I ck`
+        USE_CK=yes
+	AC_MSG_RESULT([concurrencykit found - configuring to use ck])
```
Nit:  This line is unnecessary:
```
checking for CK... yes
concurrencykit found - configuring to use ck
```

zultron commented on src/configure.ac:
```diff
@@ -976,10 +976,24 @@ AC_ARG_ENABLE(dev,
+    ],
+    [
+        USE_CK=no
+	AC_MSG_ERROR([ concurrencykit not found - please install package libck0 ])
```
This will cause the build to exit if `libck` isn't available.  Since `libck` isn't shipped for Jessie i386, can this line just be dropped, and `libck` be optional?

zultron replied:
Ah ha, turns out the version of `libck` in Jessie is 0.3.5, whereas this patch requires `PKG_CHECK_MODULES([CK], [ck >= 0.4.6], [...])`.  Backing this down to 0.3.5 and building in Jessie prints lots of errors like this:

```
hal/lib/hal_accessor.h:258:230: error: ‘ck_pr_load_double’ was not declared in this scope
```

So I think `libck` must be optional at least as long as we support Jessie.

zultron reviewed (changes requested):
Putting `libck` packages in the MK repo for Buster is worth the effort, but Jessie i386 doesn't seem like it.  I think `libck` should be made optional by removing the relevant line from this PR to work around this.

zultron commented on src/configure.ac:
```diff
@@ -976,10 +976,24 @@ AC_ARG_ENABLE(dev,
 # NML
 ##############################################################################
 
+PKG_CHECK_MODULES([CK], [ck >= 0.4.6],
```
Builds fail on Jessie, where `libck` is version 0.3.5.

zultron committed:
configure.ac:  Make `libck` optional

Also clean up extra messages

zultron committed:
debian:  Fix `libck` package dependencies

- Let `dh_shlibdeps` figure out if `libck0` is needed by binary
  packages (probably not, since the library isn't actually linked
  in (???))
- Add `libck-dev` build-dep except in Jessie (pkg too old) (missing
  Buster packages added to MK repo by @ArcEye)
